Library/Homebrew/os/mac.rb

```diff
@@ -37,6 +37,15 @@ def outdated_release?
       version < "10.9"
     end
 
+    def os_name(version = @version)
```

Comment:
In that case you can avoid making `version` a `Version` if it already is one and `10.12` doesn't need to be a `Version`.